### PR TITLE
DX-1282 Load secret data to environment variables

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>
DX-1282

---

### Preview Link

<br>

[Tyk Stack](https://deploy-preview-4493--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-stack-chart/)

[Tyk Data Plane](https://deploy-preview-4493--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-data-plane-chart/)

[Tyk OSS](https://deploy-preview-4493--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-oss-chart/)

[Tyk Control Plane](https://deploy-preview-4493--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/tyk-control-plane-chart/)

### Description

<br>

Example of how to load secret values in environment variables and access them through Tyk configuration files or API Definitions.

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

___

## **Type**
enhancement, documentation

___

## **Description**
- Added documentation on how to use Kubernetes Secrets to set environment variables in the Tyk Gateway.
- This update includes a practical example and a reference to detailed documentation, enhancing the security and flexibility of Tyk deployments in Kubernetes environments.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>tyk-stack-chart.md</strong><dd><code>Add Kubernetes Secrets Usage Example in Tyk Gateway Configuration</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md
<li>Added an example of setting an extra environment variable using <br>Kubernetes Secret data in the Tyk Gateway configuration.<br> <li> Provided a detailed explanation of how to securely inject credentials <br>into the Tyk Gateway using Kubernetes secrets.<br> <li> Included a reference link to further details on accessing secret data <br>from Tyk configuration files.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4493/files#diff-f6e45bbf530fa763d587dec798584c8eb6294229ef353d3212ab01bf6b0a1d7c">+14/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 1]
```diff
@@ -111,6 +114,32 @@ tyk-gateway:
       value: debug
 ```
 
+Example of setting extra [environment variable using ConfigMap data](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#define-container-environment-variables-using-configmap-data), using gateway as an example:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]: Added suggestion to remove duplication of example, applicable to other chart content (tyk-control-plane-chart, tyk-data-plane-chart, tyk-oss-chart, tyk-stack-chart) files

```suggestion
An example is listed below for setting extra [environment variable using ConfigMap data](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#define-container-environment-variables-using-configmap-data), using gateway:
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 2]
```diff
@@ -111,6 +114,32 @@ tyk-gateway:
...
+  gateway:
+    extraEnvs:
+    - name: CONFIG_USERNAME
+      valueFrom:
+        configMapKeyRef: 
+          name: backend-user
+          key: backend-username
+```
+
+Example of setting extra [environment variable using Secret data](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data), using gateway as an example:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
An example is listed below for setting extra [environment variable using secret data](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data), using gateway:
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 3]
```diff
@@ -111,6 +114,32 @@ tyk-gateway:
...
+  gateway:
+    extraEnvs:
+    - name: SECRET_USERNAME
+      valueFrom:
+        secretKeyRef: 
+          name: backend-user
+          key: backend-username
+```
+
+In the above example, an extra environment variable `SECRET_USERNAME` will be added to gateway container, taking value from `backend-username` value defined in the Secret `backend-user`. It is useful if you want to [access secret data from Tyk Gateway configuration file (tyk.conf) or API definitions]({{<ref "tyk-configuration-reference/kv-store#how-to-access-the-externally-stored-data">}}).
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
In the above example, an extra environment variable `SECRET_USERNAME` will be added to the Gateway container, with a value of `backend-username` associated with the secret `backend-user`. It is useful if you want to access secret data from [Tyk Gateway configuration file (tyk.conf) or API definitions]({{<ref "tyk-configuration-reference/kv-store#how-to-access-the-externally-stored-data">}}).
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks for submitting PR @caroltyk, I have made minor suggestions in Tyk Control Plane chart that are applicable to the other charts in the PR. 

@sedkis Would you be able to peer review for @caroltyk ?

**Review by sedkis — APPROVED** [review]

**Comment by carolwkc** [comment]
Thank you @dcs3spp . I have applied the suggestions to all charts.

**Review by dcs3spp — APPROVED** [review]
Many thanks @caroltyk, PR LGTM! I'll merge and release

**Pull request merged**